### ISSUE TITLE
New Rule: no-unmatchable-anb-selector

### ISSUE THREAD

**Issue by TKDev7** [eslint/css#186]
### Rule details

Disallow unmatchable An+B selectors.

### What type of rule is this?

Warns about a potential problem

### Example code

```css
a:nth-child(0n+0) {}
/*↑             ↑
 * This unmatchable An+B selector */
```

### Prior Art

```markdown
selector-anb-no-unmatchable: https://stylelint.io/user-guide/rules/selector-anb-no-unmatchable/
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

[An+B selectors](https://www.w3.org/TR/css-syntax-3/#anb-microsyntax) are one-indexed. Selectors that always evaluate to 0 will not match any elements.

**Comment by nzakas** [comment]
Sounds good to me. Marking as accepted. :+1:

**Comment by nzakas** [comment]
> Is it worth expanding the scope of this rule to detect more unmatchable selectors and naming it `no-unmatchable-selectors`?

I like that idea. It also gives us flexibility to start with a small group that we're sure of and add more over time.

@thecalamiity what do you think?

**Comment by Mouvedia** [comment]
> …but support for detecting them has been requested in Stylelint…

stylelint/stylelint#8566
stylelint/stylelint#5575

**Comment by snitin315** [comment]
@TKDev7 Are you still working on this?

**Comment by Tanujkanti4441** [comment]
Hey @TKDev7, are you still working on this, or would you like me to take it up?
